lib/braveAlerter.js

```diff
@@ -289,7 +299,29 @@ class BraveAlerter {
     } else {
       const errorMessage = `Bad request to ${request.path}: ${validationErrors.array()}`
       helpers.logError(errorMessage)
-      response.status(400).send(errorMessage)
+      response.status(400).json(errorMessage)
+    }
+  }
+
+  async handleGetActiveAlerts(request, response) {
+    const validationErrors = Validator.validationResult(request).formatWith(helpers.formatExpressValidationErrors)
+
+    if (validationErrors.isEmpty()) {
+      const alertApiKey = request.header('X-API-KEY')
+
+      const activeAlerts = await this.getActiveAlertsByAlertApiKey(alertApiKey)
+      if (activeAlerts === null) {
+        const errorMessage = `Something went wrong in request to ${request.path}. Active Alerts is null.`
+        helpers.logError(errorMessage)
+        response.status(500).json(errorMessage)
+        return
+      }
+
+      response.status(200).json(JSON.stringify(activeAlerts))
```

Comment:
this gets called to populate the home screen, and the db query is the complement to Historic Alerts, everything that's within the time threshold and has not been responded, correct?

Reply:
It's called whenever the app loads or when a push notification comes in. The actual alerts do not appear on the `HomeScreen`, they get their own special `AlertScreen` because I couldn't figure out how to make consistent-looking alerts otherwise.

Almost. The "activeAlerts" are ones that are within the time threshold and have *not been completed* (so ones that are in Responding, and Waiting for Incident Category will also be here). I think I have a comment explaining that in the implementation of `getActiveAlerts` on the Buttons and Sensors side. Since from the library's perspective, it's believing whatever the client code gives it. So if we ever want to change the definition of "activeAlerts" then this part wouldn't need to change.